### PR TITLE
Little cleanup of Quorum Cache tests

### PR DESCRIPTION
**Pull request by Donnerbart**
Motivated by warnings in the class files.

### CODE REVIEW

**Comment by Donnerbart** [comment]
run-parallel

File: hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java [hunk 1]
```diff
@@ -166,14 +160,15 @@ public void onChange(QuorumEvent quorumEvent) {
             threeNode.put(generateKeyOwnedBy(h1), "bar");
             fail();
         } catch (Exception e) {
+            EmptyStatement.ignore(e);
         }
         try {
             fourNode.put(generateKeyOwnedBy(h1), "bar");
             fail();
         } catch (Exception e) {
+            EmptyStatement.ignore(e);
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: rename the exception to expected and pleace remove the EmptyStatement.ignore. We do this for production code; but we should not do this for test code.

File: hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java [hunk 2]
```diff
@@ -240,12 +234,13 @@ public void onChange(QuorumEvent quorumEvent) {
...
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(3);
         HazelcastInstance instance1 = factory.newHazelcastInstance(config);
-        HazelcastInstance instance2 = factory.newHazelcastInstance(config);
+        factory.newHazelcastInstance(config);
         HazelcastServerCachingProvider cachingProvider = HazelcastServerCachingProvider.createCachingProvider(instance1);
         Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache(cacheName);
         try {
             cache.put(generateKeyOwnedBy(instance1), 1);
         } catch (Exception e) {
+            e.printStackTrace();
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: Don't print stack traces. Either it is a failure and the test should be aborted; or it is something which should happen and then you also don't want to see the stacktrace.

**Comment by pveentjer** [comment]
Minor review comments regarding exception handling

File: hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java (hunk 1 above)
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: IDEA still creates a warning for the empty catch. This always grabs my attention (yellow alert). Renamed the exception though.

File: hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java (hunk 2 above)
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: I wasn't sure about this one, but you're right. I'll remove the try-catch and we'll see if it crashes.

**Comment by pveentjer** [comment]
The tests could be better; one of the problem is that I see with multipl try/catch clauses. This is an indication that the test is 

action
assert (exception)
action
assert (exception)

This is a smell and normally indicates that there are multiple tests hidden in a single test method. Once that is done.. you can put an @Test(expected=blaexception.class) around it and you get rid of the try catch completely.

**Comment by Donnerbart** [comment]
I just improved the try-catch in `CacheQuorumListenerTest`. At least it's visible now that we expected a `QuorumException` all over the class. But this is not what the test is about, it is about the listener event to be fired (in case of a `QuorumException`). So it should be okay to catch the exceptions and wait for the event to be fired.

**Comment by mustafaiman** [comment]
:+1: 
these tests and map's quorum tests are very similar. it would be good to apply the same refactoring to them to be consistent.

**Comment by Donnerbart** [comment]
@pveentjer: Can you have another look? I think the tests should be okay now. The method calls which throw the exception are needed to trigger the listeners, which are the real test subject.

So it should be perfectly fine to call one or two methods before (and check that the needed exception is really thrown, since this is the happy path to the final assert).

File: hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java [hunk 3]
```diff
@@ -203,18 +204,21 @@ public boolean apply(Collection<Member> members) {
         });
         config.getCacheConfig(cacheName).setQuorumName(quorumName);
         config.addQuorumConfig(quorumConfig);
+
         HazelcastInstance instance = createHazelcastInstance(config);
         QuorumService quorumService = instance.getQuorumService();
-        Quorum quorum = quorumService.getQuorum(quorumName);
+        quorumService.getQuorum(quorumName);
```
Comment (mesutcelik) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: why are we calling this?

**Comment by mesutcelik** [comment]
:+1: with a minor comment.

File: hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java (hunk 3 above)
Comment (mustafaiman) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: i think that's a leftover. It has no use in the test.

Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/quorum/cache/CacheQuorumListenerTest.java]: Good catch, thx!

**Pull request merged**